manta-js/package/src/wallet/crate/src/lib.rs

```diff
@@ -1601,7 +1596,16 @@ impl Wallet {
                 Box::pin(async {
                     this.sign_with_transaction_data(transaction.into(), metadata.map(Into::into))
                         .await
-                        .map(SignWithTransactionDataResponse::from)
+                        .map(|response| {
+                            let posts = response
+                                .0
+                                .clone()
+                                .into_iter()
+                                .map(|x| TransferPost::from(x.0))
+                                .collect::<Vec<_>>();
+                            let txs = response.0.into_iter().map(|x| x.1).collect::<Vec<_>>();
+                            (posts, txs)
+                        })
```

Comment:
Why a tuple of vectors instead of a vector of tuples? The latter makes more sense because each `TransferPost` is linked to each `TransactionData` instance 1-to-1. 
Suggestion: 
.map(|response| {
                            response
                                .0
                                .into_iter()
                                .map(|(post, data)| (TransferPost::from(post), data))
                                .collect::<Vec<_>>();
                        })

Reply:
I pushed the changes because both @DanielZhangReal and @Apokalip agreed